enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,333 @@
+
+This enhancement would offer an optional mechanism whereby test writers have a persistent volume mounted within their test containers, provisioned, enabled, and cleaned up by scorecard.
+
+
+## Motivation
+
+Allowing scorecard tests a place to persist test output is key to running advanced tests and supporting complex test output.
+
+QE teams for example might already have tests which produce test output files, in order for them to integrate their tests into scorecard, they need a persistent volume to write test results into.
+
+This feature enables a wider adoption of tests that can be executed within scorecard.
+
+### Goals
+
+Goals of this proposal include:
+* provide a persistent volume for scorecard test developers
+* Collect persistent volume output when scorecard completes
+
+### Non-Goals
+
+The names of the persistent volumes created by scorecard for this feature
+are opaque to the test writer, meaning that they are not defined in a way
+where test writers would need or want to know the names of their volumes.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+A scorecard user can specify for a given test that they want persistent storage to be made available to that test.  This is specified within the scorecard configuration file as a PersistentVolume spec.  For example, a user would specify they want storage for a test by specifying
+in the scorecard configuration file the `storage` spec as in the following example:
+
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+    storage: 
+      spec:
+        storageClassName: hostpath
+        capacity:
+          storage: 1Gi
+        accessModes:
+          - ReadWriteOnce
+        mountPath:
+          path: /test-output
+```
+
+#### Story 2
+A scorecard user can optionally specify storage class details within the scorecard configuration file to override storage defaults. Volume size, access mode, storage class, and mount paths can be specified as follows:
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: custom
+      test: example-test
+    storage: 
+      spec:
+        storageClassName: hostpath
+        capacity:
+          storage: 1Gi
+        accessModes:
+          - ReadWriteOnce
+        mountPath:
+          path: /test-output
+```
+
+Valid values for access mode:
+ * ReadWriteOnce (default if not specified)
+ * ReadOnlyMany
+ * ReadWriteMany
```

Comment:
why do we need to let them specify the access mode?  readwriteonce seems like the only sensible value here.

readonly is useless for this purpose and readwritemany is unnecessary since it's a single pod accessing the PV.

Reply:
unless you're going to run tests in parallel each mounting a shared PV?

in which case i'd caution you that very few storage types even support RWX as an option.

Reply:
ReadWriteOnce is the default since its widely supported, in this feature the PVs are not shared by multiple tests so you are right, I could make the only valid choice ReadWriteOnce.

Reply:
updated, removed allowing access mode to be configurable.